nominal/core/dataset.py

```diff
@@ -185,6 +188,40 @@ def get_channels(
             else:
                 next_page_token = response.next_page_token
 
+    def to_pandas(self, channel_exact_match: Sequence[str] = (), channel_fuzzy_search_text: str = "") -> pd.DataFrame:
+        """Download a dataset to a pandas dataframe, optionally filtering for only specific channels of the dataset.
+        Args:
+            channel_exact_match: Filter the returned channels to those whose names match all provided strings (case insensitive).
+                For example, a channel named 'engine_turbine_rpm' would match against ['engine', 'turbine', 'rpm'],
+                whereas a channel named 'engine_turbine_flowrate' would not!
+            channel_fuzzy_search_text: Filters the returned channels to those whose names fuzzily match the provided string.
+        Returns:
+            A pandas dataframe whose index is the timestamp of the data, and column names match those of the selected channels.
+
+        Example:
+        ```
+        import nominal as nm
+
+        rid = "..." # Taken from the UI or via the SDK
+        dataset = nm.get_dataset(rid)
+        s = dataset.to_pandas()
+        print("index:", s.index, "index mean:", s.index.mean())
+        ```
+        """
+        # We concurrently download all channels for faster performance
+        futures = []
+        seriess = []
+        with ThreadPoolExecutor(max_workers=5) as pool:
+            for channel in self.get_channels(channel_exact_match, channel_fuzzy_search_text):
+                futures.append(pool.submit(channel.to_pandas))
+            if not futures:
+                raise ValueError("No channels were found given the provided filters")
+            with tqdm(total=len(futures)) as progress:
```

Comment:
Understood on the want here but to me progress bars are generally an application concern and not a library concern--maybe we can provide a progress callback or similar so we can still provide the functionality without being opinionated?

Reply:
i don't want to overcomplicate it. If you'd prefer no progress bar I'll drop it, i was just thinking about some datasets with hundreds of channels